src/App.jsx

```diff
@@ -5,8 +5,20 @@ import NFTCard from './components/NFTCard/NFTCard';
 import { Route, Routes } from 'react-router-dom';
 import Footer from "./components/Footer/Footer.jsx";
 import Collection from './components/Collection/Collection.jsx'
+import {useEffect} from 'react';
+import {useDispatch} from 'react-redux'
+import { setAddress } from './app/wallet.js';
+import ConnectMetamask from './components/ConnectMetamask.jsx';
 
 function App() {
+
+    const dispatch = useDispatch();
+    const { wallet } = ConnectMetamask();
+    useEffect(()=>{
+        if (wallet) dispatch(setAddress(wallet.accounts[0].address));
+        else dispatch(setAddress(null));
```

Comment:
here you dont need to dispatch setAddress(null) = default value of the address in redux is 0

Reply:
Shouldn't we keep this so it gets reset on log out? 
Change it to setAddress(0)?

Reply:
You are correct